pkg/instancegroups/rollingupdate_test.go

```diff
@@ -105,6 +105,30 @@ func (*erroringClusterValidator) Validate() (*validation.ValidationCluster, erro
 	return nil, errors.New("testing validation error")
 }
 
+// simulates failures in a specific node group in the map of instance groups
+type instanceGroupNodeSpecificErrorClusterValidator struct {
+	Groups    map[string]*cloudinstances.CloudInstanceGroup
+	NodeGroup string
+}
+
+func (igErrorValidator *instanceGroupNodeSpecificErrorClusterValidator) Validate() (*validation.ValidationCluster, error) {
+	instanceGroup := igErrorValidator.Groups[igErrorValidator.NodeGroup].InstanceGroup
+	if instanceGroup.IsMaster() || instanceGroup.IsBastion() {
+		instanceGroup = nil
```

Comment:
This is testing the mock, not the code under test.

Reply:
Would it be better if we include the Validate logic in the Validate mock? I thought just mocking the failures should have been enough to test this. Since we are mostly concerned about the failures return by Validate() and not how it returns it. Am I understanding something wrongly?

Reply:
If it was the rolling update code that was responsible for ensuring failures for IGs of role master still blocked rolling update, then testing that in the rolling updates code would make sense. In the current design, where it is the validation code that is responsible for that logic, that logic should be tested by the validation unit tests, not the rolling update unit tests.

Reply:
Fair enough. This makes sense. I ll push all logic to determine whether we should wait or not for validation to the rolling update code rather than the validation code. It would make things much easier and simpler